### PR TITLE
update gazebo article

### PR DESCRIPTION
**Pull request by khancyr**
Rebase from https://github.com/ArduPilot/ardupilot_wiki/pull/480
Based on work from @squilter

### CODE REVIEW

**Comment by khancyr** [comment]
@lvale update with correction for ubuntu 14.04 . Gazebo wiki is more outdated than apm one ! ;-P . My error was that I already have ROS install so I got most gazebo dependancies already install...

**Comment by magicrub** [comment]
@squilter does this look good to you? If so I'll merge it in

File: dev/source/docs/using-gazebo-simulator-with-sitl.rst [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+Build and install Gazebo
+::
+
+    cd ~
+    hg clone https://bitbucket.org/osrf/gazebo
+    cd gazebo
+    hg checkout ardupilot
+    mkdir build
+    cd build
+    cmake ../
```
Comment (lvale) [inline dev/source/docs/using-gazebo-simulator-with-sitl.rst]: Hi.
Fails here 

```
-- Looking for cos in m - found
-- ----------------------------------------

-- BUILD WARNINGS
-- 	HDF5 not found
-- 	DART not found, for dart physics engine option, please install libdart-core4-dev.
-- 	Player not found, gazebo plugin for player will not be built.
-- 	Bullet > 2.82 not found, for bullet physics engine option, please install libbullet2.82-dev.
-- 	libusb-1.0 not found. USB peripherals support will be disabled.
-- 	Oculus Rift support will be disabled.
-- 	GDAL not found, Digital elevation terrains support will be disabled.
-- 	ronn not found, manpages won't be generated
-- 	Graphviz not found, Model editor's schematic view will be disabled.
-- 	OpenSource Virtual Reality (OSVR) support will be disabled.
-- END BUILD WARNINGS

-- BUILD ERRORS: These must be resolved before compiling.
-- 	Missing: Ignition msgs0 library.
-- 	Missing: libqwt-dev. Required for plotting.
-- END BUILD ERRORS

CMake Error at CMakeLists.txt:294 (message):
  Errors encountered in build.  Please see the BUILD ERRORS above.

```

Comment (lvale) [inline dev/source/docs/using-gazebo-simulator-with-sitl.rst]: [CMakeOutput.log.txt](https://github.com/ArduPilot/ardupilot_wiki/files/531804/CMakeOutput.log.txt)
[CMakeError.log.txt](https://github.com/ArduPilot/ardupilot_wiki/files/531803/CMakeError.log.txt)

**Comment by khancyr** [comment]
WIP . I retesting it on a VM with ubuntu 14.04... it is killing my computer ... I don't think it will finish building gazebo a day... That is why I wanted to wait a release from osrf but as many people a dealing problem with old docs ...

**Comment by khancyr** [comment]
Ok depencies seems to be fixed and build is launch... will give a status report before go to sleep ... If my computer doesn't die before :-S

File: dev/source/docs/using-gazebo-simulator-with-sitl.rst [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+Build and install Gazebo
+::
+
+    hg clone https://bitbucket.org/osrf/gazebo ~/gazebo_ws/gazebo
+    cd ~/gazebo_ws/gazebo
+    hg checkout ardupilot
+    mkdir build
+    cd build
+    cmake ../
+    make -j4 # NOTE: This will take a long time!
```
Comment (lvale) [inline dev/source/docs/using-gazebo-simulator-with-sitl.rst]: @khancyr
on the build Gazebo step it fails here:
cmake ../
this is the report
```
-- BUILD WARNINGS
-- HDF5 not found
-- DART not found, for dart physics engine option, please install libdart-core4-dev.
-- Player not found, gazebo plugin for player will not be built.
-- Bullet > 2.82 not found, for bullet physics engine option, please install libbullet2.82-dev.
-- libusb-1.0 not found. USB peripherals support will be disabled.
-- Oculus Rift support will be disabled.
-- GDAL not found, Digital elevation terrains support will be disabled.
-- ronn not found, manpages won't be generated
-- Graphviz not found, Model editor's schematic view will be disabled.
-- OpenSource Virtual Reality (OSVR) support will be disabled.
-- END BUILD WARNINGS
-- Custom CFlags: -std=c++11 -Wall -Wextra -Wno-long-long -Wno-unused-value -Wfloat-equal -Wshadow -Winit-self -Wswitch-default -Wmissing-include-dirs -pedantic -fvisibility=hidden -fvisibility-inlines-hidden -std=c++11
-- Build Type: RelWithDebInfo
-- Install path: /usr/local
-- Configuration successful. Type make to compile gazebo
-- Configuring done
-- Generating done
-- Build files have been written to: /home/parallels/gazebo_ws/gazebo/build
parallels@ubuntu:~/gazebo_ws/gazebo/build$
```

File: dev/source/docs/using-gazebo-simulator-with-sitl.rst [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+as an external simulator for Copter.
+
+Overview
+========
+
+Gazebo is a well-known and respected robotics simulator. We will be compiling 
+Gazebo from source, because no current release has built-in support for ArduCopter.
+
+.. warning::
+   Gazebo support is still under work on October 2016. The following instructions may be outdated or thing could be broken. In those case, open an issue on ardupilot github or discuss.
```
Comment (squilter) [inline dev/source/docs/using-gazebo-simulator-with-sitl.rst]: maybe rephrase "Gazebo support is still under development, as of October 2016. If it seems that these instructions are outdated, please open an issue on the ardupilot_wiki github."

File: dev/source/docs/using-gazebo-simulator-with-sitl.rst [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+Preconditions
+=============
+
+We recommend Ubuntu starting from 14.04.2 (it is working on 16.04 but untest on 16.10) as this is the platform used for testing
+this approach and is also known to be compatible with SITL.
+
+Compiling and installing Gazebo
+===============================
+
+We will be using a standard version of ArduPilot, but a custom fork of Gazebo until gazebo plugin get merge into Gazebo build.
```
Comment (squilter) [inline dev/source/docs/using-gazebo-simulator-with-sitl.rst]: "We will be using a standard version of ArduPilot but a custom fork of Gazebo, until the gazebo plugin gets merge into Gazebo-master."

**Comment by squilter** [comment]
The diff looks messed up for `sitl-simulator-software-in-the-loop.rst`. It looks like every line was removed and then re-added. Otherwise one or two grammar changes that I noted.

**Comment by khancyr** [comment]
@squilter Thanks for english correction ^^ ! I don't understand the mess in sitl-simulator-software-in-the-loop.rst as I only modify one line. It could be an encoding or lineending change ?(I use linux, but it could have been done previously on windows). Sorry for bypassing your PR but I don't acheived to solve the merge conflict on your PR, so it was simpler to start again 
I update some dependencies again, the build with be longer but it will include Bullet engine (from last_letter simulation) and trigger less warning .
I didn't have time to look at Opengl error that lvale get ... My VM crash before the error trigger, but it could be related to VM . 
I will try on a fresh ubuntu 16.04 and another computer with VM to check if everything is good now.

**Comment by OXINARF** [comment]
@khancyr @squilter It's most likely line endings changing. You can remove white space changes by adding `?w=1` to the end of the URL, like https://github.com/ArduPilot/ardupilot_wiki/pull/572/files?w=1

File: dev/source/docs/using-gazebo-simulator-with-sitl.rst [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+-------------------------------
+
+We will also need to get a gazebo model of a quadcopter.
+
+::
+
+    hg clone https://bitbucket.org/osrf/gazebo_models ~/gazebo_ws/gazebo_models
+    cd ~/gazebo_ws/gazebo_models
+    hg checkout zephyr_demos
+    echo 'export GAZEBO_MODEL_PATH=~/gazebo_models' >> ~/.bashrc
```
Comment (khancyr) [inline dev/source/docs/using-gazebo-simulator-with-sitl.rst]: error here

**Comment by squilter** [comment]
Once this patch is in, let's not forget to close
#251
#438
#439

**Comment by hamishwillee** [comment]
@lvale @squilter Anyone tested this to the point they are happy to accept it?

**Comment by magicrub** [comment]
Status update?

**Comment by squilter** [comment]
The instructions worked fine for me. But that's because I wrote them. Should probably get at least one other person to follow the instructions successfully first.

**Comment by magicrub** [comment]
I'll give it a few more days then merge. Maybe we can add some text that says "This was only tested by the author, please post an issue if it doesn't work as described" with a link?

**Comment by khancyr** [comment]
@magicrub @squilter It works for me too even on fresh ubuntu 16.04. I didn't retry on 14.04. Lvale test were failing in VM but I think it is related to VM. A test on fresh ubuntu 14.04 (not in VM) could be great. 

Normally, I add a warning stating that it is still under works. There will be some PR on gazebo side in the next days, but I will update the wiki here according to the changes and also add what it working and what is not !
